lib/index.js

```diff
@@ -95,21 +95,24 @@ class Parser extends Transform {
       }
     }
     // Normalize option `delimiter`
-    if(options.delimiter === undefined || options.delimiter === null || options.delimiter === false){
-      options.delimiter = Buffer.from(',')
-    }else if(typeof options.delimiter === 'string' && options.delimiter.length !== 0){
-      options.delimiter = Buffer.from(options.delimiter)
-    }else if(
-      (Buffer.isBuffer(options.delimiter) && options.delimiter.length === 0) ||
-      (typeof options.delimiter === 'string' && options.delimiter.length === 0) ||
-      (!Buffer.isBuffer(options.delimiter) && typeof options.delimiter !== 'string')
-    ){
-      throw new CsvError('CSV_INVALID_OPTION_DELIMITER', [
-        'Invalid option delimiter:',
-        'delimiter must be a non empty string or buffer,',
-        `got ${JSON.stringify(options.delimiter)}`
-      ])
-    }
+    let delimiter = options.delimiter
+    let delimiter_error = new CsvError('CSV_INVALID_OPTION_DELIMITER', [
```

Comment:
If the error is thrown in 2 places, then I create 2 distincts errors, I can't take care of this later. It is good that you check the array length, I was thinking about this as well.

Reply:
okay I will add 2 distinct errors

Reply:
the reason why I put it in one place is because after changing the options.delimiter, I still need access to the original options.delimiter for showing it inside the thrown error

Reply:
The reason I did this is because I needed access to the original delimiter option for throwing it inside the loop. If i change the delimiter to array, then I lose access to the original one and hence the error generated does not match the expected error.

Reply:
It make sense then